### PR TITLE
feat(NumberInput): add experimental mobile variant

### PR DESCRIPTION
**Pull request by emyarod**
Closes IBM/carbon-components-react#1981

This PR matches the number input component to the experimental spec

#### Changelog

**New**

- experimental mobile variant

### CODE REVIEW

**Comment by emyarod** [comment]
`carbon-components` version update required after https://github.com/IBM/carbon-components/pull/2003 is merged in

File: src/components/NumberInput/NumberInput.js [hunk 1]
```diff
@@ -217,6 +221,7 @@ class NumberInput extends Component {
       allowEmpty,
       innerRef: ref,
       translateWithId: t,
+      mobile,
```
Comment (joshblack) [inline src/components/NumberInput/NumberInput.js]: Should we do `isMobile`?

**Comment by shixiedesign** [comment]
Hey Andrew, I noticed this in both vanilla and react versions: (apologies I didn't catch it in the last num input PR!!)
- [ ] Light, mobile variant buttons are missing hover states. Hover color should be `$hover-ui`
num-input

**Comment by emyarod** [comment]
blocked by https://github.com/IBM/carbon-components/pull/2027

**Comment by asudoh** [comment]
Removing `blocked` label as #2027 has been merged.

**Review by asudoh — COMMENTED** [review]
LGTM, just one thing wrt the prop name - Thanks @emyarod!

**Comment by emyarod** [comment]
bumped dependency version since IBM/carbon-components#2027 has been merged

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @emyarod!

**Review by shixiedesign — APPROVED** [review]
Looks good! Thank you! 🎉

**Review by alisonjoseph — APPROVED** [review]
🎉

**Pull request merged**